### ISSUE TITLE
Inaccessible Monkey Menu

### ISSUE THREAD

**Issue by jsl303** [greasemonkey/greasemonkey#2794]
I'm a visually impaired user, and Monkey Menu seemed to become inaccessible since since 4.0 with screen reader.
Is there a way to bring back Monkey Menu under tools menu?
Or, is there way to present the UI objects in html view?
Thanks!

**Comment by Sxderp** [comment]
> Is there a way to bring back Monkey Menu under tools menu?

Unfortunately no.

Accessibility is being worked on (#2639) and feedback is appreciated. Being an individual who doesn't require these types of assistance can make it difficult to understand what will and will not work. As new updates are rolled out please continue to post your thoughts and feedback.

@arantius , If the Monkey Menu does not work then the decision to remove the full page menu may have to be reversed. Of course this is still early and the initial changes haven't been pushed yet.

**Comment by arantius** [comment]
>  Is there a way to bring back Monkey Menu under tools menu?

Maybe?  Partially?

> Monkey Menu seemed to become inaccessible since since 4.0

You're not the only person to report this, we've made the first bit of progress towards improving this recently ( https://github.com/greasemonkey/greasemonkey/commit/056f9788730b69c2c0ff046ec1cce43db3112f25 ).  I'll keep this issue open to report the next beta build which will include that change, so you can let us know how much this improves things for you.

**Comment by arantius** [comment]
The fix in PR #2660 is packaged in version 4.2beta2:
https://addons.mozilla.org/firefox/downloads/file/833159/greasemonkey-4.2beta2-an+fx.xpi?src=devhub

This is only a start, but better than we've done so far! Testing is always appreciated!

My experience so far is: click the toolbar button (I don't know how a non-sighted/non-mouse-using person does this???), press tab, then up/down/enter mostly work.  The first "down" selects the second item, not the first.

**Comment by Eselce** [comment]
The first `up` after pressing `tab` selects first item, as I can see...

**Comment by Sxderp** [comment]
I purposely made it so the first down selected the second item while the first up selects the first. 

It felt 'normal' to me. I've seen that setup before but I don't recall where.

**Comment by jsl303** [comment]
Wow, this is awesome!!!!!!
After I click the Monkey Menu and press tab, I was able to select a 
script using up/down arrows.
Then I was able to choose edit, uninstall, and so on using down/up arrows!
A couple of more suggestions...
Could you attach a shortcut to bring that menu? It would be a lot easier 
than trying to navigate to toolbar, and click monkey menu using a screen 
reader.
Also, I can't edit scripts using the editor Greasemonkey uses.
Right now, I just select all, cut, paste into an external editor, edit, 
paste back into the Monkey editor.
For some reason, when I press arrows, screen reader can't read what's 
selected. Screen reader thinks it's text edit field, but it doesn't read 
the content inside.
Would you mind looking into making the monkey editor accessible? Or, is 
there a way to set Greasemonkey to open an external editor when I select 
edit?
Thanks!!!!!!

**Comment by Sxderp** [comment]
> Could you attach a shortcut to bring that menu?

I did a quick search and I this is what I found in regard to APIs.
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/browserAction/openPopup
> Activating a keyboard shortcut defined by the extension (note: this is not currently supported in Firefox).

I don't know if they're working on a fix.

> Or, is there a way to set Greasemonkey to open an external editor

Not at the moment, but when the backend script installation / update flow is reworked and cleaned up to be faster and act more like 3.x, it should be easier to add an interface for external programs. Assuming it's added it won't be as simple as filesystem access (rip Addons) but a python wrapper or somesuch could potentially be used.

**Comment by arantius** [comment]
I'm glad you find this an improvement.  We know we're not done, further things like the editor are being tracked at #2639.
